packages/dd-trace/src/appsec/iast/analyzers/hardcoded-password-analyzer.js

```diff
@@ -0,0 +1,22 @@
+'use strict'
+
+const { HARDCODED_PASSWORD } = require('../vulnerabilities')
+const HardcodedBaseAnalyzer = require('./hardcoded-base-analyzer')
+
+const allRules = [
+  {
+    id: 'hardcoded-password',
+
+    // eslint-disable-next-line max-len
+    regex: /(?:pwd|pswd|pass|secret)(?:[0-9a-z\-_\t.]{0,20})(?:[\s|']|[\s|""]){0,3}(?:=|>|:{1,3}=|\|\|:|<=|=>|:|\?=)(?:'|""|\s|=|\x60){0,5}([0-9a-z\-_.=]{10,150})(?:['"\s\x60;]|$)/i,
```

Comment:
The regexes are inherited from gitleaks, I've tried to keep them unchanged. But as you say, that part could be simplified. 
So, what do you prefer? :P

Reply:
> what do you prefer?

what I prefer doesn't matter. The question is, does it worth?

Reply:
As you said, we know `=` is always in our `key=value` string so I don't know if in terms of performance there would be much difference between `(?:=|>|:{1,3}=|\|\|:|<=|=>|:|\?=)` or `=` because the first char that this block is searching for is `=` so it will end quickly with a match

```
  (?:                            =                        |>|:{1,3}=|\|\|:|<=|=>|:|\?=)
   |                             |
non-capturing block        first attempt always present
```

that's why I preferred to keep the original regex

Reply:
ok, let's maintain it